### DISCUSSION TITLE
Ongoing improvements to the Readme file

### DISCUSSION THREAD

**Discussion by kaveh808** [kaveh808/kons-9#13] [General]
The Readme file is our chance to make a first impression. I imagine us continually improving and tweaking it.

I have just edited it, so please have a look and let me know what could be better.

Please pay special attention to the onboarding section for new team members. We want to make that as easy as possible.

**Comment by vidjuheffex** [comment]
Some thoughts, keeping in mind I don't come from the CL world, so take this with a grain of salt:

1) I needed to (require "asdf"), unblocking myself with all of this took all of 30 seconds, so not a big deal. But including:

```
(require "asdf")
(push #p"<your directory>/kons-9/" asdf:*central-registry*)
```
in this initial snippet wouldn't hurt any.

2) the next thing I ran into was `(ql:quickload ...)`. Again, a quick google search later got me sorted out, (and I don't think kons-9 is responsible for teaching me how to install quicklisp) but I do think some text like: `prerequistes: QuickLisp needs to be installed` with a link to quicklisp wouldn't hurt.

3) Those hangups aside, I found it fairly easy to get to get the Viewport window appearing on screen.  The next step however hung me up:

"Open demo.lisp and start evaluating the blocks of code. Things should appear in the graphics window."

Where? How? I know I'm lacking some foundational CL stuff, and am diving in the deep end here sort-to-speak, so maybe this question will seem dumb to me after I woodshed on CL this weekend, but *my guess* at the moment is this referring to SLIME or some-emacs CL thing, if so should I have been running all the aforementioned code in Emacs? So maybe a little more details on the workflow would help new users.

**Reply by kaveh808** [reply]
If you running ccl in a shell, cut and paste `demo.lisp` code blocks (usually starting with `(with-clear-and-redraw`) into the lisp prompt. This will execute the code.

I do have some CL lessons at: [https://github.com/kaveh808/Kaveh-Common-Lisp-Lessons](https://github.com/kaveh808/Kaveh-Common-Lisp-Lessons). The file `lesson-00.pdf` has instructions for setting up emacs and slime.

Let me know how it goes. I appreciate you taking the effort to learn this. Hope you fall in love with CL, just like we have!

**Comment by mtreis86** [comment]
Using \*central-repository\* is sorta deprecated at this point. From the manual,
"Please do not use the central-registry if you are a novice, and do not instruct novices to use the central-registry."
Manual suggested procedure for adding locations https://asdf.common-lisp.dev/asdf.html#Configuring-ASDF-to-find-your-systems

**Reply by omarbassam88** [reply]
i use `(push (uiop:getcwd) ql:*local-project-directories*)` This works flawlessly with quick load if QuickLisp is already set up to load automatically with your repl. I also don't need the `(require asdf)` line.

**Reply by kaiwulf** [reply]
Yes, `(require "asdf")` gives me NIL.
The suggestion you gave worked, however.

**Reply by mtreis86** [reply]
`require` is implementation-dependent so it may be best to not use it. All modern CL implementations provide ASDF anyways

**Reply by kaveh808** [reply]
If someone wants to update this in the Readme file, that's cool. I went with the simplest thing that worked.

**Reply by WillAdams** [reply]
Unfortunately, it's been a _very_ long while since I did anything w/ Lisp, and I'm finding myself quite lost.

Managed to get it installed (both on Windows and on Mac)

Tried navigating to: /Users/~/Documents/kons-9-main/

and launching the program there.

running the suggested command:

    (push (uiop:getcwd) ql:*local-project-directories*)

triggered an error that  uiop wasn't loaded --- @vidjuheffex 's suggestion of:

    (require "asdf")

got me past that hurdle, but I'm still getting an error trying to move beyond that. I'd suggest a set of at least two walkthroughs, one for Windows, the other for Unix/MacOS which explains everything in excruciating detail --- or a start-up script which loads things just by double-clicking on it.

**Reply by kaveh808** [reply]
I apologize for the hassles, Will. We definitely need improved documentation.  I'll open an issue and ask for volunteers.

Have you been able to get the system running yet?

**Reply by kaveh808** [reply]
Opened issue #175

**Reply by WillAdams** [reply]
No, haven't made any progress since writing up the above.

**Comment by kaveh808** [comment]
Bear with us. We are working on detailed instructions: #175
